src/js/view/controls/settings-menu.js

```diff
@@ -148,12 +126,52 @@ export function setupSubmenuListeners(settingsMenu, controlbar, model, api) {
             provider.setPlaybackRate.bind(model.getVideo()),
             model.get('playbackRate')
         );
+    };
+
+    model.change('mediaModel', (newModel, mediaModel) => {
+        // Quality Levels
+        mediaModel.change('levels', onQualitiesChanged, this);
+        mediaModel.on('change:currentLevel', (changedModel, currentQuality) => {
+            activateSubmenuItem('quality', currentQuality);
+        }, this);
+
+        // Audio Tracks
+        mediaModel.change('audioTracks', onAudiotracksChanged.bind(this, mediaModel), this);
+        mediaModel.on('change:currentAudioTrack', (changedModel, currentAudioTrack) => {
+            activateSubmenuItem('audioTracks', currentAudioTrack);
+        }, this);
     }, this);
 
+    // Captions
+    model.change('captionsList', onCaptionsChanged, this);
+    model.change('captionsIndex', (changedModel, index) => {
+        const captionsSubmenu = settingsMenu.getSubmenu('captions');
+        if (captionsSubmenu) {
+            captionsSubmenu.activateItem(index);
+            controlbar.toggleCaptionsButtonState(!!index);
+        }
+    }, this);
+
+    // Playback Rates
+    model.change('playbackRates', onPlaybackRatesChange, this);
     model.change('playbackRate', (changedModel, playbackRate) => {
         const rates = model.get('playbackRates');
         if (rates) {
             activateSubmenuItem('playbackRates', rates.indexOf(playbackRate));
         }
     }, this);
+
+    // Remove the audio tracks, qualities, and playback rates submenus when casting
+    model.on('change:castActive', (changedModel, active) => {
```

Comment:
This should be `model.change`. We I connect and then disconnect, the quality submenu shows up